contracts/voting-verifier/src/contract.rs

```diff
@@ -64,3 +65,781 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+                mock_info(worker.address.as_str(), &[]),
+                msg,
+            );
+            assert!(res.is_ok());
+        });
+
+        let msg = ExecuteMsg::EndPoll {
+            poll_id: 1u64.into(),
+        };
+
+        let res = execute(
+            deps.as_mut(),
+            mock_env_expired(),
+            mock_info(SENDER, &[]),
+            msg,
+        );
+        assert!(res.is_ok());
+
+        let res: Vec<(CrossChainId, VerificationStatus)> = from_binary(
+            &query(
+                deps.as_ref(),
+                mock_env(),
+                QueryMsg::GetMessagesStatus {
+                    messages: messages.clone(),
+                },
+            )
+            .unwrap(),
+        )
+        .unwrap();
+        assert_eq!(
+            res,
+            vec![
+                (
+                    messages[0].cc_id.clone(),
+                    VerificationStatus::SucceededOnChain
+                ),
+                (messages[1].cc_id.clone(), VerificationStatus::FailedOnChain),
+                (messages[2].cc_id.clone(), VerificationStatus::NotFound),
+                (
+                    messages[3].cc_id.clone(),
+                    VerificationStatus::FailedToVerify
+                )
+            ]
+        );
+
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info(SENDER, &[]),
+            msg_verify,
+        );
+        assert!(res.is_ok());
+
+        let res: Vec<(CrossChainId, VerificationStatus)> = from_binary(
+            &query(
+                deps.as_ref(),
+                mock_env(),
+                QueryMsg::GetMessagesStatus {
+                    messages: messages.clone(),
+                },
+            )
+            .unwrap(),
+        )
+        .unwrap();
+        assert_eq!(
+            res,
+            vec![
+                (
+                    messages[0].cc_id.clone(),
+                    VerificationStatus::SucceededOnChain
+                ),
+                (messages[1].cc_id.clone(), VerificationStatus::FailedOnChain),
+                (messages[2].cc_id.clone(), VerificationStatus::InProgress),
+                (messages[3].cc_id.clone(), VerificationStatus::InProgress)
+            ]
+        );
+    }
+
+    #[test]
+    fn should_query_message_statuses() {
+        let mut deps = setup();
+
+        let messages = messages(10);
+        let msg = ExecuteMsg::VerifyMessages {
+            messages: messages.clone(),
+        };
+
+        let res = execute(deps.as_mut(), mock_env(), mock_info(SENDER, &[]), msg).unwrap();
+
+        let reply: VerifyMessagesResponse = from_binary(&res.data.unwrap()).unwrap();
+
+        assert_eq!(reply.verification_statuses.len(), messages.len());
+        assert_eq!(
+            reply.verification_statuses,
+            messages
+                .iter()
+                .map(|message| (message.cc_id.clone(), VerificationStatus::None))
```

Comment:
do you think it's confusing that an unknown message return `None` instead of `InProgress` after verification has started?

Reply:
to be honest I would remove the whole response `data`, it is being set to the status before the verification starts, I think the original purpose was to do some logic with it in the aggregate verifier to combine multiple verifiers, I'm not sure it makes sense to use the previous status before the verification starts, but anyway I think the real combining logic would end up in the query not the execute since that's what the gateway uses.

Reply:
yeah I was wandering about that too